Userland/Libraries/LibGUI/VimEditingEngine.cpp

```diff
@@ -708,11 +708,13 @@ Optional<TextPosition> VimMotion::get_position(VimEditingEngine& engine)
             // above in get_range normalizes some values which shouldn't be
             // end-exclusive during normal operations.
             bool is_at_start = range.end().column() == 0;
+            auto& line = editor.line(range.end().line());
+
             size_t column = is_at_start ? 0 : range.end().column() - 1;
+            column = min(column, line.length() - !in_visual_mode);
```

Comment:
I think this is easier to read and understand if made explicit:

```suggestion
            column = min(column, line.length() - (in_visual_mode ? 0 : 1));
```